### PR TITLE
[WIP] mingw 32/64

### PR DESCRIPTION
**Pull request by drewwells**
This formula will add mingw 32/64 bit to Homebrew. To do so, gcc needs to be compiled with mingw target, mingw headers, and mingw crt.

Mingw provides a minimalistic gnu environment for other architectures. With this, Windows 32/64 binaries can be compiled on OS X. Windows is required to run these binaries.

Shell script that builds mingw: https://gist.github.com/drewwells/ee5d42dc29b70b7b61af

### CODE REVIEW

**Comment by mikemcquaid** [comment]
I think a better option would may be to add this as a `with` option to the existing `gcc` formula.

**Comment by drewwells** [comment]
@mikemcquaid I started down that route, can you install multiple `with` versions of the same formula?

GCC needs to be built with mingw 32bit and 64bit targets. The will likely already be using gcc with their host target too. I played with swapping out with options and homebrew said I already installed gcc...

**Comment by mikemcquaid** [comment]
@drewwells You can't but you can definitely compile GCC with support for multiple guests beyond just the host.

**Comment by joelself** [comment]
:thumbsup:

**Comment by mikemcquaid** [comment]
@drewwells Any thoughts on my last comment?

**Comment by drewwells** [comment]
Sorry, I've dropped the ball on this. @mikemcquaid I made the switch to brew provided gcc here: https://github.com/drewwells/homebrew/commit/f3af2367600a6374124078cd6385c715af76dd61

I got stuck on errors about missing target platform headers.

**Comment by drewwells** [comment]
```
==> ../mingw-w64-crt/configure --host=x86_64-w64-mingw32 --prefix=/private/tmp/mingw20160226-81320-1hy37qj/mingw-w64-v3.1.0/x86_64-w64-ming
Last 15 lines from /Users/drew/Library/Logs/Homebrew/mingw/05.configure:
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking _mingw_mac.h usability... no
checking _mingw_mac.h presence... no
checking for _mingw_mac.h... no
configure: error: Please check if the mingw-w64 header set and the build/host option are set properly.
```

File: Library/Formula/mingw.rb [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  #option "without-fortran", "Build without the gfortran compiler"
+  # enabling multilib on a host that can't run 64-bit results in build failures
+  #option "without-multilib", "Build without multilib support" if MacOS.prefer_64_bit?
+
+  depends_on "gmp4"
+  depends_on "gcc48" => ["with-all-languages"]
+  depends_on "binutils"
+  depends_on "isl011"
+  depends_on "isl012"
+  depends_on "cloog018"
```
Comment (mikemcquaid) [inline Library/Formula/mingw.rb]: None of these formula exist in Homebrew core.

Comment (drewwells) [inline Library/Formula/mingw.rb]: Ming only compiles with these specific versions.

Comment (mikemcquaid) [inline Library/Formula/mingw.rb]: They'll need to be vendored in this formula or this submitted to homebrew-versions instead.

Comment (drewwells) [inline Library/Formula/mingw.rb]: What do you mean by vendorered? I've never vendored something in homebrew
before

Comment (mikemcquaid) [inline Library/Formula/mingw.rb]: Installed with a `resource` rather than a dependency.

Comment (bfontaine) [inline Library/Formula/mingw.rb]: Ping?

**Pull request closed without merging**